gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/apim/core/plugin/use_case/GetResourcePluginUseCase.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.apim.core.plugin.use_case;
+
+import io.gravitee.apim.core.UseCase;
+import io.gravitee.apim.core.plugin.crud_service.ResourcePluginCrudService;
+import io.gravitee.apim.core.plugin.domain_service.PluginFilterByLicenseDomainService;
+import io.gravitee.apim.core.plugin.model.ResourcePlugin;
+import io.gravitee.apim.core.plugin.query_service.ResourcePluginQueryService;
+import io.gravitee.rest.api.service.exceptions.ResourceNotFoundException;
+import java.util.Set;
+import lombok.AllArgsConstructor;
+
+@UseCase
+@AllArgsConstructor
+public class GetResourcePluginUseCase {
+
+    private final ResourcePluginCrudService resourcePluginCrudService;
+    private final ResourcePluginQueryService resourcePluginQueryService;
+    private final PluginFilterByLicenseDomainService licenseChecker;
+
+    public Output execute(Input input) {
+        var resourcePlugin =
+            this.licenseChecker.setPluginDeployedStatusDependingOnLicense(
+                    Set.of(resourcePluginCrudService.findById(input.resourceId)),
+                    input.organizationId
+                )
+                .stream()
+                .findFirst()
+                .orElseThrow(() -> new ResourceNotFoundException(input.resourceId));
```

Comment:
what about something like (considering my comment about Optional has been respected, else wrap with Optional.of():

```suggestion
            resourcePluginCrudService.findById(input.resourceId)
              .map(resource -> license.checker.setPluginDeployedStatusDependingOnLicense(Set.of(resource), input.organizationId)
                .orElseThrow(() -> new ResourceNotFoundException(input.resourceId));
```

Reply:
🚀 but maybe more with a flatMap 🤔